onesignal-public.php

```diff
@@ -89,9 +35,11 @@ public static function onesignal_header()
         } ?>
     <?php
     if (defined('ONESIGNAL_DEBUG') && defined('ONESIGNAL_LOCAL')) {
-        echo '<script src="https://localhost:3001/sdks/OneSignalSDK.js" async></script>';
+        wp_register_script('local_sdk', plugins_url('https://localhost:3001/sdks/OneSignalSDK.js', __FILE__), array('jquery'), false, true);
+        wp_enqueue_script('local_sdk');
     } else {
-        echo '<script src="https://cdn.onesignal.com/sdks/OneSignalSDK.js" async></script>';
+        wp_register_script('remote_sdk', plugins_url('https://cdn.onesignal.com/sdks/OneSignalSDK.js', __FILE__), array('jquery'), false, true);
```

Comment:
We need to keep `async` in the `<script>` tag. Unfortunately `wp_register_script` does seem to support the async flag. Found a work around to still get it added even when using `wp_register_script` though.
https://stackoverflow.com/questions/18944027/how-do-i-defer-or-async-this-wordpress-javascript-snippet-to-load-lastly-for-fas

Reply:
Great catch!